Update docs for deploying Fleet on Kubernetes (#4755)
fleetdm/fleet: docs/Deploying/Server-Installation.md

## Deploying Fleet on Kubernetes

In this guide, we're going to focus on deploying Fleet only on a Kubernetes cluster. Kubernetes is a container orchestration tool that was open sourced by Google in 2014.

We will assume you have `kubectl` and MySQL and Redis are all set up and running. Optionally you have minikube to test your deployment locally on your machine.

To deploy the Fleet server and connect to its dependencies(MySQL and Redis) we will set up a `deployment.yml` file with the following specifications:

```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: fleet-deployment
  labels:
    app: fleet
spec:
  replicas: 3
  selector:
    matchLabels:
      app: fleet
  template:
    metadata:
      labels:
        app: fleet
    spec:
      containers:
      - name: fleet

>         image: fleetdm/fleet

Reviewer: This should specify a version which we then update with the current release of Fleet as time goes on. Otherwise it pulls "latest" which could be anything.

As part of a wider discussion, do we have a way to render variables like current release of Fleet into our docs?

Reviewer 2: > As part of a wider discussion, do we have a way to render variables like current release of Fleet into our docs?

We don't. This would be a nice problem to solve, but is potentially tricky due to these variables being built into multiple "targets" (eg. fleetdm.com/docs and the [Helm charts](https://github.com/fleetdm/fleet/blob/main/.github/workflows/release-helm.yaml))

I agree that a version needs to be specified here. For now (until we solve the above problem), @DominusKelvin please update step 1 of https://fleetdm.com/docs/contributing/releasing-fleet to also point to where this needs to be changed for the Kubernetes documentation.

Author: Okay this means I should both update the image to point to a particular version of Fleet?

Reviewer 2: Yes

Author: Added.

>         image: fleetdm/fleet:4.12.0

Reviewer 3: ```suggestion
        image: fleetdm/fleet:4.13.2
```

Can you please update the example file to use the latest Fleet version? Thanks :)

        env:

>           - name: FLEET_MYSQL_ADDRESS

Reviewer: I don't think literally everything needs to be a secret personally. It doesn't hurt anything except readability though.

Author: Yeah I think having this as the default convention makes for good practice in my opinion in terms of not having secrets in code.

            valueFrom:
              secretKeyRef:
                name: fleet_secrets
                key: mysql_address
          - name: FLEET_MYSQL_DATABASE
            valueFrom:
              secretKeyRef:
                name: fleet_secrets
                key: mysql_database
          - name: FLEET_MYSQL_PASSWORD
            valueFrom:
              secretKeyRef:
                name: fleet_secrets
                key: mysql_password
          - name: FLEET_MYSQL_USERNAME
            valueFrom:
              secretKeyRef:
                name: fleet_secrets
                key: mysql_username
          - name: FLEET_REDIS_ADDRESS
            valueFrom:
              secretKeyRef:
                name: fleet_secrets
                key: redis_address
        resources:
          requests:
            memory: "64Mi"
            cpu: "250m"
          limits:
            memory: "128Mi"
            cpu: "500m"
        ports:
        - containerPort: 3000

```

> Notice we are using secrets to pass in values for Fleet's dependencies' environment variables.

Reviewer 3: ```suggestion
> This example file uses secrets to pass in values for Fleet's dependencies.
```

I think we can turn this sentence into a note (add the ">" character). What do you think?

I also suggested removed the "environment variables" section from the sentence for clarity. Is the sentence still accurate with this section removed?
